apps/happy-blocks/block-library/pricing-plans/view.tsx

```diff
@@ -31,6 +32,10 @@ function renderPricingPlansBlock( element: HTMLElement | Element ) {
 }
 
 domReady( () => {
+	recordTracksEvent( 'calypso_happyblocks_upgrade_plan_view', {
```

Comment:
I think doing this here will record the event every single time this file is loaded, despite the block is used or not.

To confirm my understanding, can you try this branch on a page which doesn't have any instance of this block? If I'm correct the event will be firing anyways, despite there's no usage of it.


If it is fine to record N events, if there are N usages of the block in the page (suppose the block is used on a Topic's body and then again in the replies) then you could move this to `renderPricingPlansBlock `. That function won't run if there are no usages.

Reply:
> I think doing this here will record the event every single time this file is loaded, despite the block is used or not.
> To confirm my understanding, can you try this branch on a page which doesn't have any instance of this block? If I'm correct the event will be firing anyways, despite there's no usage of it.

I have tried it while developing and was also part of the testing instructions, and it is not the case, the [documentation](https://developer.wordpress.org/block-editor/reference-guides/block-api/block-metadata/#view-script) is a bit vague, but I think it makes sense to not call all blocks' view scripts all the time.

> If it is fine to record N events, if there are N usages of the block in the page (suppose the block is used on a Topic's body and then again in the replies) then you could move this to renderPricingPlansBlock . That function won't run if there are no usages.

I thought about this a bit, my thinking was more in the context of page views so I went with counting a single time per page no matter how many blocks are there, but if it makes more sense we can track per block, cc: @naxoc

Reply:
Makes sense then! Thanks for double checking it.